### PR TITLE
Bump to mono:2019-08

### PR DESCRIPTION
**Pull request by directhex**

### CODE REVIEW

**Comment by directhex** [comment]
Keep an eye on this one, all our integration documentation is apparently no longer applicable. I hope I updated the right places.

File: build-tools/xaprepare/xaprepare/Application/Abi.cs [hunk 1]
```diff
@@ -26,12 +26,12 @@ partial class Abi
 			new Abi { Name = AbiNames.HostJit.Win32,           Type = AbiType.HostJit,   Is64Bit = false, IsCross = false, IsHost = true,  IsLlvm = false, IsWindows = true },
 			new Abi { Name = AbiNames.HostJit.Win64,           Type = AbiType.HostJit,   Is64Bit = true,  IsCross = false, IsHost = true,  IsLlvm = false, IsWindows = true },
 
-			new Abi { Name = AbiNames.CrossAot.ArmV7a,         Type = AbiType.CrossAot,  Is64Bit = false, IsCross = true,  IsHost = false, IsLlvm = false, IsWindows = false },
```
Comment (jonpryor) [inline build-tools/xaprepare/xaprepare/Application/Abi.cs]: These changes shouldn't be needed as of bb7ffb775599bdb4675301ae06d1bbc8e8ff8566 and https://github.com/xamarin/xamarin-android/blob/aef33486566f117f0089ea5f4cbebc37aa37ff95/build-tools/xaprepare/xaprepare/Application/MonoCrossRuntime.cs#L21

Please revert the changes to this file.

**Comment by jonpryor** [comment]
Squash-and-merge commit message:

```
Changes: http://github.com/mono/mono/compare/2c3aeaf3780de7392a0d3cbe4dcf86846eb4dffa...29b1ac19c961b959a09097dbc0fe4cd567cc5298

	$ git diff --shortstat 2c3aeaf3..29b1ac19
	 1528 files changed, 45421 insertions(+), 21967 deletions(-)

Changes: https://github.com/mono/api-doc-tools/compare/d03e819838c6241f92f90655cb448cc47c9e8791...5da8127af9e68c9d58a90aa9de21f57491d81261

	$ git diff --shortstat d03e8198..5da8127a
	 1001 files changed, 86168 insertions(+), 11863 deletions(-)

Changes: https://github.com/mono/api-snapshot/compare/e09042dad3a4cd2e96e790adadd355b0e0c9772d...1ca8e82f51c5f988027e80b59aa59f7748fd0448

	$ git diff --shortstat e09042da..1ca8e82f
        28 files changed, 612 insertions(+), 217 deletions(-)

Changes: https://github.com/mono/cecil/compare/a6c8f5e1070d25c38cbfaa415d6f6853e5236687...cb6c1ca9aaef869841bfc3e2e7407b81b979b875

	$ git diff --shortstat a6c8f5e1..cb6c1ca9
	 13 files changed, 233 insertions(+), 88 deletions(-)

Changes: https://github.com/mono/corefx/compare/4806207f6fcf66a4c7f901f8479b82e488201f0c...470e0e10310639395d5ac2ee1652635cac9df658

	$ git diff --shortstat 4806207f...470e0e10
	 4 files changed, 31 insertions(+), 12 deletions(-)

Changes: https://github.com/mono/linker/compare/ebe2a1f4b989f5a21c44b4f3fe5631286aee5539...1f87de35a914e0d263a7fc44cd20b5a5c3d040d3

	$ git diff --shortstat ebe2a1f4...1f87de35
        90 files changed, 3219 insertions(+), 1224 deletions(-)

Upstream-Fixes: https://bugs.winehq.org/show_bug.cgi?id=47561
Upstream-Fixes: https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems/edit/967582
Upstream-Fixes: https://github.com/dotnet/coreclr/issues/25071
Upstream-Fixes: https://github.com/dotnet/coreclr/issues/25242
Upstream-Fixes: https://github.com/dotnet/coreclr/issues/25632
Upstream-Fixes: https://github.com/dotnet/coreclr/issues/25709
Upstream-Fixes: https://github.com/dotnet/corefx/issues/37955
Upstream-Fixes: https://github.com/dotnet/corefx/issues/38455
Upstream-Fixes: https://github.com/mono/mono/issues/7377
Upstream-Fixes: https://github.com/mono/mono/issues/8747
Upstream-Fixes: https://github.com/mono/mono/issues/9621
Upstream-Fixes: https://github.com/mono/mono/issues/9664
Upstream-Fixes: https://github.com/mono/mono/issues/9706
Upstream-Fixes: https://github.com/mono/mono/issues/10201
Upstream-Fixes: https://github.com/mono/mono/issues/10645
Upstream-Fixes: https://github.com/mono/mono/issues/10748
Upstream-Fixes: https://github.com/mono/mono/issues/10848
Upstream-Fixes: https://github.com/mono/mono/issues/12141
Upstream-Fixes: https://github.com/mono/mono/issues/13311
Upstream-Fixes: https://github.com/mono/mono/issues/13408
Upstream-Fixes: https://github.com/mono/mono/issues/13412
Upstream-Fixes: https://github.com/mono/mono/issues/13891
Upstream-Fixes: https://github.com/mono/mono/issues/13923
Upstream-Fixes: https://github.com/mono/mono/issues/13945
Upstream-Fixes: https://github.com/mono/mono/issues/14170
Upstream-Fixes: https://github.com/mono/mono/issues/14214
Upstream-Fixes: https://github.com/mono/mono/issues/14214
Upstream-Fixes: https://github.com/mono/mono/issues/14215
Upstream-Fixes: https://github.com/mono/mono/issues/14243
Upstream-Fixes: https://github.com/mono/mono/issues/14377
Upstream-Fixes: https://github.com/mono/mono/issues/14495
Upstream-Fixes: https://github.com/mono/mono/issues/14555
Upstream-Fixes: https://github.com/mono/mono/issues/14724
Upstream-Fixes: https://github.com/mono/mono/issues/14729
Upstream-Fixes: https://github.com/mono/mono/issues/14772
Upstream-Fixes: https://github.com/mono/mono/issues/14792
Upstream-Fixes: https://github.com/mono/mono/issues/14793
Upstream-Fixes: https://github.com/mono/mono/issues/14809
Upstream-Fixes: https://github.com/mono/mono/issues/14830
Upstream-Fixes: https://github.com/mono/mono/issues/14839
Upstream-Fixes: https://github.com/mono/mono/issues/14841
Upstream-Fixes: https://github.com/mono/mono/issues/14847
Upstream-Fixes: https://github.com/mono/mono/issues/14864
Upstream-Fixes: https://github.com/mono/mono/issues/14871
Upstream-Fixes: https://github.com/mono/mono/issues/14917
Upstream-Fixes: https://github.com/mono/mono/issues/14945
Upstream-Fixes: https://github.com/mono/mono/issues/14946
Upstream-Fixes: https://github.com/mono/mono/issues/14948
Upstream-Fixes: https://github.com/mono/mono/issues/14957
Upstream-Fixes: https://github.com/mono/mono/issues/14959
Upstream-Fixes: https://github.com/mono/mono/issues/14960
Upstream-Fixes: https://github.com/mono/mono/issues/14961
Upstream-Fixes: https://github.com/mono/mono/issues/14963
Upstream-Fixes: https://github.com/mono/mono/issues/14971
Upstream-Fixes: https://github.com/mono/mono/issues/14972
Upstream-Fixes: https://github.com/mono/mono/issues/14975
Upstream-Fixes: https://github.com/mono/mono/issues/15023
Upstream-Fixes: https://github.com/mono/mono/issues/15048
Upstream-Fixes: https://github.com/mono/mono/issues/15058
Upstream-Fixes: https://github.com/mono/mono/issues/15080
Upstream-Fixes: https://github.com/mono/mono/issues/15182
Upstream-Fixes: https://github.com/mono/mono/issues/15188
Upstream-Fixes: https://github.com/mono/mono/issues/15189
Upstream-Fixes: https://github.com/mono/mono/issues/15261
Upstream-Fixes: https://github.com/mono/mono/issues/15262
Upstream-Fixes: https://github.com/mono/mono/issues/15263
Upstream-Fixes: https://github.com/mono/mono/issues/15265
Upstream-Fixes: https://github.com/mono/mono/issues/15268
Upstream-Fixes: https://github.com/mono/mono/issues/15307
Upstream-Fixes: https://github.com/mono/mono/issues/15324
Upstream-Fixes: https://github.com/mono/mono/issues/15329
Upstream-Fixes: https://github.com/mono/mono/issues/15330
Upstream-Fixes: https://github.com/mono/mono/issues/15441
Upstream-Fixes: https://github.com/mono/mono/issues/15446
Upstream-Fixes: https://github.com/mono/mono/issues/15503
Upstream-Fixes: https://github.com/mono/mono/issues/15541
Upstream-Fixes: https://github.com/mono/mono/issues/15551
Upstream-Fixes: https://github.com/mono/mono/issues/15556
Upstream-Fixes: https://github.com/mono/mono/issues/15596
Upstream-Fixes: https://github.com/mono/mono/issues/15691
Upstream-Fixes: https://github.com/mono/mono/issues/15692
Upstream-Fixes: https://github.com/mono/mono/issues/15740
Upstream-Fixes: https://github.com/mono/mono/issues/15751
Upstream-Fixes: https://github.com/mono/mono/issues/15760
Upstream-Fixes: https://github.com/mono/mono/issues/15781
Upstream-Fixes: https://github.com/mono/mono/issues/15794
Upstream-Fixes: https://github.com/mono/mono/issues/15825
Upstream-Fixes: https://github.com/mono/mono/issues/15853
Upstream-Fixes: https://github.com/mono/mono/issues/15878
Upstream-Fixes: https://github.com/mono/mono/issues/15887
Upstream-Fixes: https://github.com/mono/mono/issues/15990
Upstream-Fixes: https://github.com/mono/mono/issues/16032
Upstream-Fixes: https://github.com/mono/mono/issues/16411
Upstream-Fixes: https://github.com/mono/mono/issues/16486
Upstream-Fixes: https://github.com/mono/mono/issues/25709
Upstream-Fixes: https://github.com/mono/mono/issues/38821
Upstream-Fixes: https://github.com/xamarin/xamarin-android/issues/3112
Upstream-Fixes: https://github.com/xamarin/xamarin-android/issues/3168
```
